src/client/actions/MessageUpdate.js

```diff
@@ -32,6 +32,29 @@ class MessageUpdateAction extends Action {
       updated: null,
     };
   }
+
+  patchDataPacket(data, message) {
+    data.type = data.hasOwnProperty('type') ? data.type : Constants.MessageTypes.indexOf(message.type);
```

Comment:
With permission from @Gawdl3y I have changed Constants.MessageTypes to an array. (I had been talking with Gus and others about this, because there was no reason for it to be an Object.) Once travis finishes, this review won't be valid anymore. (but github won't be able to recognize it)